### PR TITLE
Allow batch size of zero in row format JSON

### PR DESCRIPTION
**Pull request by peakji**
Fixes https://github.com/tensorflow/serving/issues/1610

### CODE REVIEW

File: tensorflow_serving/util/json_tensor_test.cc [hunk 1]
```diff
@@ -1017,6 +1017,51 @@ TEST(JsontensorTest, FromJsonMultipleNamedTensorsErrors) {
...
+      dim { size: 2 }
+    }
+    )",
+                                          &tensormap["str_tensor"]));
+
+  ASSERT_TRUE(TextFormat::ParseFromString(R"(
+    dtype: DT_FLOAT
+    tensor_shape {
+      dim { }
+      dim { size: 1 }
```
Comment (lilao) [inline tensorflow_serving/util/json_tensor_test.cc:1053]: Nit: can you change this size 1 to a different number, to verify that we only require the first dimension to match?

File: tensorflow_serving/util/json_tensor_test.cc [hunk 2]
```diff
@@ -1017,6 +1017,51 @@ TEST(JsontensorTest, FromJsonMultipleNamedTensorsErrors) {
...
+    }
+    )",
+                                          &tensormap["float_tensor"]));
+
+  string json;
+  TF_EXPECT_OK(
+      MakeJsonFromTensors(tensormap, JsonPredictRequestFormat::kRow, &json));
+  TF_EXPECT_OK(CompareJson(json, R"({ "predictions": [] })"));
+}
+
```
Comment (lilao) [inline tensorflow_serving/util/json_tensor_test.cc:1064]: Can you also add a test when the first dimension of the first tensor is 0, and that of the second tensor is greater than 0?

File: tensorflow_serving/util/json_tensor_test.cc (hunk 1 above)
Comment (peakji) [reply tensorflow_serving/util/json_tensor_test.cc]: Updated in https://github.com/tensorflow/serving/pull/1615/commits/1f7aebd906a70ba0fa04105ceee6227960b764f7

File: tensorflow_serving/util/json_tensor_test.cc (hunk 2 above)
Comment (peakji) [reply tensorflow_serving/util/json_tensor_test.cc:1064]: New test case added in https://github.com/tensorflow/serving/pull/1615/commits/91afd42dab8ce50f86bbf65065dce0c28163314b

**Comment by peakji** [comment]
@lilao Thanks for reviewing my PR, hope we can make it into the `v2.2.0` release, cheers!

**Pull request merged**